internal/api/v1/application/stage.go

```diff
@@ -339,50 +303,125 @@ func validateBlob(ctx context.Context, blobUID string, app models.AppRef, s3Conn
 	return nil
 }
 
-// newPipelineRun is a helper which creates a Tekton pipeline run
+// newJobRun is a helper which creates a Job
 // resource from the given staging params
-func newPipelineRun(app stageParam) *v1beta1.PipelineRun {
-	str := v1beta1.NewArrayOrString
+func newJobRun(app stageParam) *batchv1.Job {
+
+	// fake stage params of the previous to pull the old image url from.
+	previous := app
+	previous.Stage = models.NewStage(app.PreviousStageID)
 
 	protocol := "http"
 	if app.S3ConnectionDetails.UseSSL {
 		protocol = "https"
 	}
-	awsScript := fmt.Sprintf("aws --endpoint-url %s://$1 s3 cp s3://$2/$3 $(workspaces.source.path)/$3", protocol)
-	awsArgs := []string{
-		app.S3ConnectionDetails.Endpoint,
-		app.S3ConnectionDetails.Bucket,
-		app.BlobUID,
+
+	awsScript := fmt.Sprintf("aws --endpoint-url %s://%s s3 cp s3://%s/%s /workspace/source/%s",
+		protocol, app.S3ConnectionDetails.Endpoint, app.S3ConnectionDetails.Bucket,
+		app.BlobUID, app.BlobUID)
+
```

Comment:
With the old implementation only tar.gz seems to work when doing the 'push' form the ui. When supply a zip file the `/store` and `/stage` calls both succeed however the call to `/complete` fails with a 500 error
```
Server Backtrace: Tasks Completed: 4 (Failed: 1, Cancelled 0), Skipped: 0\ngithub.com/epinio/epinio/internal/api/v1/application.Controller.Staged.func1\n\t/go/src/github.com/epinio/internal/api/v1/application/stage.go:285
```
It would help the usability of the UI if both tar.gz and zip file types were supported.

Supporting zip would also make it easier to support the 'folder' source as well (it's much easier client side to create a zip file from a local folder).

Reply:
Created new issue - https://github.com/epinio/epinio/issues/1149 - In the icebox.